Meta/lint-python.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+set -e
+
+script_path=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+cd "${script_path}/.." || exit 1
+
+if ! command -v flake8 >/dev/null 2>&1 ; then
+    echo "flake8 is not available. Either skip this script, or install flake8."
+    exit 1
```

Comment:
It's fine to do `exit 1` here.
It's fine to run this as part of `lint-ci.sh`.
It's fine to run `lint-ci.sh` as part of `git commit`.
However, all three at the same time sounds like a bad idea.

We could:
- Require devs to install flake8 if they want to use the pre-commit hook
- Add an envvar `SKIP_MISSING` or something that is set for the pre-commit hook, and results in an `exit 0` here.
- Eat cupcakes? Dunno :)

Reply:
`lint-prettier.sh` does the exact same thing 🤔 . So does `lint-clang-format.sh`, if the linter tool is not installed. I would say the options are more along the lines of...:

- Make all linters required to use the pre-commit hook
- Make no linters required to use the pre-commit hook
- Make just clang-format required to use the pre-commit hook

Reply:
I think the most sensible option is to have the scripts check if there are any files to process first, and only then complain about the absence of the required commands.

- Someone who never touches JS files should not be required to install prettier (or flake8 for that matter). The pre-commit hook already passes staged files to these scripts to ensure they don't check all files on every commit
- CI always checks all files (git ls-files), but installing all required tools trivial